### DISCUSSION TITLE
Add a feature "OnReceive" to can make a progress bar

### DISCUSSION THREAD

**Discussion by mmvisualgit** [geby/synapse#6] [General]
I have add in "httpsend.pas" and "blcksock.pas" the feature "OnReceive". With this event I can set a prograssbar to see the download state.

I have tested ist, please change this files in the git repro.

Thank you very much.

[httpsend.zip](https://github.com/geby/synapse/files/14350848/httpsend.zip)

**Comment by geby** [comment]
Why not use common TBlockSocket.OnStatus event? It can be used for progressbars on any socket and for all protocols.

**Comment by rvk01** [comment]
> I have add in "httpsend.pas" and "blcksock.pas" the feature "OnReceive". With this event I can set a prograssbar to see the download state.

So you made a OnReceive but not an OnSend? Don't you need a progressbar for sending too?

Anyway... like @geby mentioned, there is already a TBlockSocket.OnStatus.
See example here: http://synapse.ararat.cz/doku.php/public:howto:onstatus
and here: http://synapse.ararat.cz/doku.php/public:howto:ftp_progressbar

You could just hook into THTTPSend.Sock.OnStatus.
Works for sending too ;)

**Comment by mmvisualgit** [comment]
In the "procedure TBlockSocket.RecvStreamSize" is a while, there I have include the "OnReceive".

I test the "TBlockSocket.OnStatus".

**Comment by rvk01** [comment]
> In the "procedure TBlockSocket.RecvStreamSize" is a while, there I have include the "OnReceive".

RecvStreamSize -> RecvBufferStr -> RecvBufferEx -> RecvPacket -> RecvBuffer
And RecvBuffer has the DoMonitor and DoStatus.
The DoStatus/OnStatus can be used for status messages, the DoMonitor/OnMonitor can be used for the actual data.
